boards/FSM/Src/App/App_Brake.c

```diff
@@ -1,77 +1,141 @@
     bool (*is_brake_actuated)(void);
+
+    bool (*front_pressure_sensor_ocsc)(void);
+    bool (*rear_pressure_sensor_ocsc)(void);
+    bool (*pedal_travel_sensor_ocsc)(void);
 };
 
 struct Brake *App_Brake_Create(
-    float (*get_pressure_psi)(void),
-    bool (*is_pressure_sensor_open_or_short_circuit)(void),
-    bool (*is_brake_actuated)(void),
-    float min_pressure_psi,
-    float max_pressure_psi)
+    float (*get_front_pressure_psi)(void),
+    bool (*front_pressure_sensor_ocsc)(void),
+    float (*get_rear_pressure_psi)(void),
+    bool (*rear_pressure_sensor_ocsc)(void),
+    float (*get_pedal_travel)(void),
+    bool (*pedal_travel_sensor_ocsc)(void),
+    bool (*is_brake_actuated)(void))
 {
     struct Brake *brake = malloc(sizeof(struct Brake));
     assert(brake != NULL);
 
-    brake->pressure_in_range_check = App_InRangeCheck_Create(get_pressure_psi, min_pressure_psi, max_pressure_psi);
-    brake->min_pressure_psi        = min_pressure_psi;
-    brake->max_pressure_psi        = max_pressure_psi;
-    brake->get_pressure_psi        = get_pressure_psi;
-    brake->is_pressure_sensor_open_or_short_circuit = is_pressure_sensor_open_or_short_circuit;
-    brake->is_brake_actuated                        = is_brake_actuated;
+    brake->front_pressure_in_range_check =
+        App_InRangeCheck_Create(get_front_pressure_psi, MIN_BRAKE_PRESSURE_PSI, MAX_BRAKE_PRESSURE_PSI);
+    brake->rear_pressure_in_range_check =
+        App_InRangeCheck_Create(get_rear_pressure_psi, MIN_BRAKE_PRESSURE_PSI, MAX_BRAKE_PRESSURE_PSI);
+    brake->get_front_pressure_psi = get_front_pressure_psi;
+    brake->get_rear_pressure_psi  = get_rear_pressure_psi;
+    brake->get_pedal_travel       = get_pedal_travel;
+    brake->is_brake_actuated      = is_brake_actuated;
+
+    brake->front_pressure_sensor_ocsc = front_pressure_sensor_ocsc;
+    brake->rear_pressure_sensor_ocsc  = rear_pressure_sensor_ocsc;
+    brake->pedal_travel_sensor_ocsc   = pedal_travel_sensor_ocsc;
 
     return brake;
 }
-
 void App_Brake_Destroy(struct Brake *brake)
 {
-    App_InRangeCheck_Destroy(brake->pressure_in_range_check);
+    App_InRangeCheck_Destroy(brake->front_pressure_in_range_check);
+    App_InRangeCheck_Destroy(brake->rear_pressure_in_range_check);
     free(brake);
 }
 
-struct InRangeCheck *App_Brake_GetPressureInRangeCheck(const struct Brake *const brake)
-{
-    return brake->pressure_in_range_check;
-}
-
+// act
 bool App_Brake_IsBrakeActuated(const struct Brake *brake)
 {
     return brake->is_brake_actuated();
 }
 
-bool App_Brake_IsPressureSensorOpenOrShortCircuit(const struct Brake *brake)
+// pressures
+float App_Brake_GetFrontPSI(const struct Brake *brake)
+{
+    return brake->get_front_pressure_psi();
+}
+float App_Brake_GetRearPSI(const struct Brake *brake)
 {
-    return brake->is_pressure_sensor_open_or_short_circuit();
+    return brake->get_rear_pressure_psi();
+}
+bool App_Brake_PressureElectricalFault(const struct Brake *brake)
```

Comment:
lets make these separate functions. Helps us find the electrical issue faster in the event of the fault being thrown

Reply:
seperate CAN messages too?

Reply:
yes please